### PR TITLE
add CloudEvent enum

### PR DESCRIPTION
**Pull request by linuxbasic**

### CODE REVIEW

File: cloudevents/src/lib.rs [hunk 1]
```diff
@@ -100,6 +104,28 @@ let json = serde_json::to_string(&event.unwrap()).unwrap();
...
 
+## Deserialization
+
+```
+use serde_json;
+use cloudevents::cloudevent_v1_0;
+use cloudevents::{Data, CloudEvent};
+
+let data = "{\"type\":\"test type\",\"specversion\":\"1.0\",\"source\":\"http://www.google.com\",\"id\":\"id\",\"datacontenttype\":\"application/json\",\"data\":\"\\\"test\\\"\"}";
+let expected_event = cloudevent_v1_0!(
```
Comment (Lazzaretti) [inline cloudevents/src/lib.rs:120]: use the new `cloudevent!`

File: cloudevents/src/lib.rs [hunk 2]
```diff
@@ -100,6 +104,28 @@ let json = serde_json::to_string(&event.unwrap()).unwrap();
...
+  event_type: "test type",
+  source: "http://www.google.com",
+  event_id: "id",
+  datacontenttype: "application/json",
+  data: Data::from_string("\"test\""),
+).unwrap();
+
+match serde_json::from_str(data).unwrap() {
+  CloudEvent::V1_0(event) => assert_eq!(event, expected_event),
+  _ => {}
```
Comment (Lazzaretti) [inline cloudevents/src/lib.rs]:
```suggestion
  _ => panic!(),
```

**Review by Lazzaretti — CHANGES_REQUESTED** [review]

File: cloudevents/src/lib.rs [hunk 3]
```diff
@@ -81,7 +81,11 @@ let event : Result<CloudEventV0_2, Error> = cloudevent_v0_2!(
 );
 ```
 
-# To serialize the event as JSON, just use `serde_json`
+# JSON encoding
+
+The CloudEvents can be serialized/deserialized with `serde_json`
+
+## Serialization
```
Comment (Lazzaretti) [inline cloudevents/src/lib.rs:93]: the first example should use the `cloud_event!` macro

File: cloudevents/src/lib.rs [hunk 4]
```diff
@@ -100,6 +104,28 @@ let json = serde_json::to_string(&event.unwrap()).unwrap();
...
+  datacontenttype: "application/json",
+  data: Data::from_string("\"test\""),
+).unwrap();
+
+match serde_json::from_str(data).unwrap() {
+  CloudEvent::V1_0(event) => assert_eq!(event, expected_event),
+  _ => {}
+}
+```
+
```
Comment (Lazzaretti) [inline cloudevents/src/lib.rs:135]: update the readme with the `cargo readme`

**Review by Lazzaretti — CHANGES_REQUESTED** [review]

File: cloudevents/src/lib.rs (hunk 1 above)
Comment (linuxbasic) [reply cloudevents/src/lib.rs:120]: we expect v1.0

File: README.md [hunk 5]
```diff
@@ -44,6 +28,24 @@ let event : Result<CloudEventV1_0, Error> = cloudevent_v1_0!(
 );
```
Comment (Lazzaretti) [inline README.md:31]: add also a full example, with all fields

File: README.md [hunk 6]
```diff
@@ -44,6 +28,24 @@ let event : Result<CloudEventV1_0, Error> = cloudevent_v1_0!(
...
+    .source("http://www.google.com")
+    .event_type("test type")
+    .datacontenttype("application/json")
+    .data(Data::from_string("\"test\""))
+    .build()
+    .unwrap()
+);
+```
 
 ## Usage with spec version 0.2
```
Comment (Lazzaretti) [inline README.md:53]: please also wrap the CloudEventBuilder::v0_2 into an CloudEvent::V0_2

**Review by Lazzaretti — CHANGES_REQUESTED** [review]

**Review by Lazzaretti — APPROVED** [review]

**Pull request merged**